### ISSUE TITLE
`sort_pub_dependencies` false positive

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#2271]
Repro:

```yaml
name: animations
description: Fancy pre-built animations that can easily be integrated into any Flutter application.
version: 1.1.1
homepage: https://github.com/flutter/packages/tree/master/packages/animations

environment:
  sdk: ">=2.2.2 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

dev_dependencies:
  flutter_test:
    sdk: flutter
  vector_math: ^2.0.8
```

https://github.com/flutter/packages/blob/837b1a321d3f00034556ef5b385abb93c0843715/packages/animations/pubspec.yaml

See failure: https://cirrus-ci.com/task/5329170379046912

/fyi @dnfield

**Comment by pq** [comment]
See also: https://github.com/PhilipsHue/flutter_reactive_ble/pull/120

**Comment by pq** [comment]
Hmmmm... I'm not getting this to repro locally.  Looking at the reported failure, I wonder if this wasn't actually a mis-report?

The test failure was flagging `example/pubspec.yaml`

![image](https://user-images.githubusercontent.com/67586/95516559-439ab200-0974-11eb-921f-ace7f35b4359.png)

which is in fact not sorted:

https://github.com/flutter/packages/blob/837b1a321d3f00034556ef5b385abb93c0843715/packages/animations/example/pubspec.yaml

Whereas we were looking at the root pubspec above (and that seems ok).

🤔 

/cc @dnfield

**Comment by dnfield** [comment]
Ah! you are right.

I will file a fix for the animations package.

**Comment by dnfield** [comment]
https://github.com/flutter/packages/pull/220

**Comment by pq** [comment]
Sweet!

**Comment by pq** [comment]
When I un-revert (and for future reference), is there an easy way for me to test pro-actively if an engine roll would be blocked by a given SDK commit?
